### PR TITLE
Draft: Restructure User Settings 

### PR DESCRIPTION
**Pull request by InternetRamen**
Solves #3127

### CODE REVIEW

File: app/assets/stylesheets/components/_tabs.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
+.tab-container {
+  list-style: none;
+  padding: 0;
+  display: flex;
+  border-bottom: 1px solid $muted;
```
Comment (cjdenio) [inline app/assets/stylesheets/components/_tabs.scss]:
```suggestion
  border-bottom: 0.5px solid map-get($palette, slate);
  background-color: $dark;
  z-index: 5;
```

Updated to match the underline on the "Settings" header

File: app/assets/stylesheets/components/_forms.scss [hunk 2]
```diff
@@ -48,10 +48,30 @@ select {
```
Comment (cjdenio) [inline app/assets/stylesheets/components/_forms.scss]: I think this deserves its own PR

File: app/views/users/_settings_nav.html.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<ul class="tab-container">
+    <li class="<%= 'active' if active == 'general'  %>">
+        <%= link_to '/my/settings/', data:{turbo: true} do %>
+            General
+        <% end %>
```
Comment (cjdenio) [inline app/views/users/_settings_nav.html.erb]:
```suggestion
        <%= link_to 'General', '/my/settings/', data: { turbo: true } %>
```

File: app/views/users/_settings_nav.html.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    </li>
+    <li class="<%= 'active' if active == 'previews' %>">
+        <%= link_to '/my/settings/previews', data:{turbo: true} do %>
+            <span class="xs-hide">Feature</span> Previews
+        <% end %>
+    </li>
+    <li class="<%= 'active' if active == 'security' %>">
+        <%= link_to '/my/settings/security', data:{turbo: true} do %>
+            Security
+        <% end %>
```
Comment (cjdenio) [inline app/views/users/_settings_nav.html.erb]:
```suggestion
        <%= link_to 'Security', '/my/settings/security', data: { turbo: true } %>
```

**Comment by InternetRamen** [comment]
Does anyone know how to create link_to buttons with form elements? Or should I just create a whole new form for updating these settings.

File: app/assets/stylesheets/components/_tabs.scss [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  li {
+    padding: 0.5rem 1rem;
+    color: $muted;
+    cursor: pointer;
```
Comment (cjdenio) [inline app/assets/stylesheets/components/_tabs.scss]:
```suggestion
```

Since `<a>` already has this

**Review by cjdenio — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/components/_forms.scss (hunk 2 above)
Comment (InternetRamen) [reply app/assets/stylesheets/components/_forms.scss]: Yup

File: app/views/users/edit.html.erb [hunk 6]
```diff
@@ -6,324 +6,121 @@
...
-      Sign out everywhere else
-    <% end %>
-  </div>
-  <div class="grid grid--split">
-    <% @sessions.order(created_at: :desc).each do |session| %>
-      <div class="card border<%= ' b--warning' if session.impersonated? %><%= ' b--success' if current_session.id == session.id %>">
-        <% if session.impersonated? %>
-          <span class="admin-tools">Impersonated by <%= user_mention session.impersonated_by %></span>
+        <div class="field">
+        <%= form.label :profile_picture, class: 'fileupload' %>
```
Comment (cjdenio) [inline app/views/users/edit.html.erb]:
```suggestion
```

**Review by cjdenio — APPROVED** [review]

**Pull request merged**